WordPress/WordPressApi/WordPressOrgXMLRPCApi.swift

```diff
@@ -0,0 +1,274 @@
+    }
+
+    //MARK: - Network requests
+
+    /**
+     Executes a XMLRPC call for the method specificied with the arguments provided.
+
+     - parameter method:  the xmlrpc method to be invoked
+     - parameter parameters: the parameters to be encoded on the request
+     - parameter success:    callback to be called on successful request
+     - parameter failure:    callback to be called on failed request
+
+     - returns:  a NSProgress object that can be used to track the progress of the request and to cancel the request. If the method
+     returns nil it's because something happened on the request serialization and the network request was not started, but the failure callback
+     will be invoked with the error specificing the serialization issues.
+     */
+    public func callMethod(method: String,
+                           parameters: [AnyObject]?,
+                           success: SuccessResponseBlock,
+                           failure: FailureReponseBlock) -> NSProgress?
+    {
+        //Encode request
+        let request: NSURLRequest
+        do {
+            request = try requestWithMethod(method, parameters: parameters)
+        } catch let encodingError as NSError {
+            failure(error: encodingError, httpResponse: nil)
+            return nil
+        }
+
+        // Create task
+        let task = session.dataTaskWithRequest(request) { (data, urlResponse, error) in
+            do {
+                let responseObject = try self.handleResponseWithData(data, urlResponse: urlResponse, error: error)
+                success(responseObject: responseObject, httpResponse: urlResponse as? NSHTTPURLResponse)
+            } catch let error as NSError {
+                failure(error: error, httpResponse: urlResponse as? NSHTTPURLResponse)
+                return
+            }
+        }
+        task.resume()
+        return createProgresForTask(task)
+    }
+
+    /**
+     Executes a XMLRPC call for the method specificied with the arguments provided, by streaming the request from a file.
+     The locaiton of the file used to stream the request is the one provided in usingFileURLForCache parameter.
+
+     - parameter method:  the xmlrpc method to be invoked
+     - parameter parameters: the parameters to be encoded on the request
+     - parameter usingFileURLForCache: the file URL where to store the request data to sent to the server
+     - parameter success:    callback to be called on successful request
+     - parameter failure:    callback to be called on failed request
+
+     - returns:  a NSProgress object that can be used to track the progress of the request and to cancel the request. If the method
+     returns nil it's because something happened on the request serialization and the network request was not started, but the failure callback
+     will be invoked with the error specificing the serialization issues.
+     */
+    public func callStreamingMethod(method: String,
```

Comment:
The name implies that the method itself is streaming, with sounds weird. And following [Swift API Guidelines](https://swift.org/documentation/api-design-guidelines/#omit-needless-words), I think we should rename both methods:
- `callMethod(_:parameters:success:failure:)` -> `call(method:parameters:success:failure:)`
- `callStreamingMethod(_:parameters:success:failure:)` -> `streamingCall(method:parameters:success:failure:)`